core/templates/dev/head/pages/community-dashboard-page/community-dashboard-page.controller.ts

```diff
@@ -0,0 +1,105 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the community dashboard page.
+ */
+
+require('base_components/BaseContentDirective.ts');
+require(
+  'components/common-layout-directives/common-elements/' +
+  'background-banner.directive.ts');
+require(
+  'pages/community-dashboard-page/translation-opportunities/' +
+  'translation-opportunities.directive.ts');
+require(
+  'pages/community-dashboard-page/voiceover-opportunities/' +
+  'voiceover-opportunities.directive.ts');
+
+require('domain/utilities/LanguageUtilService.ts');
+require('domain/utilities/UrlInterpolationService.ts');
+
+require(
+  'pages/community-dashboard-page/community-dashboard-page.constants.ajs.ts');
+
+angular.module('oppia').directive('communityDashboardPage', [
+  'UrlInterpolationService', function(
+      UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/community-dashboard-page/' +
+      'community-dashboard-page.directive.html'),
+      controllerAs: '$ctrl',
+      controller: [
+        '$window', 'LanguageUtilService', 'TranslationLanguageService',
+        'COMMUNITY_DASHBOARD_TABS_DETAILS',
+        'DEFAULT_OPPORTUNITY_LANGUAGE_CODE',
+        function(
+            $window, LanguageUtilService, TranslationLanguageService,
+            COMMUNITY_DASHBOARD_TABS_DETAILS,
+            DEFAULT_OPPORTUNITY_LANGUAGE_CODE) {
+          var ctrl = this;
+          var LAST_SELECTED_TRANSLATION_LANGUAGE = (
+            'last_selected_translation_lang');
+          var prevLanguageCode = $window.localStorage.getItem(
+            LAST_SELECTED_TRANSLATION_LANGUAGE);
+          var allAudioLanguageCodes = LanguageUtilService
+            .getAllVoiceoverLanguageCodes();
+
+          ctrl.languageCodesAndDescriptions = (
+            allAudioLanguageCodes.map(function(languageCode) {
+              return {
+                id: languageCode,
+                description: (
+                  LanguageUtilService.getAudioLanguageDescription(
+                    languageCode))
+              };
+            }));
+          ctrl.languageCode = (
+          allAudioLanguageCodes.indexOf(prevLanguageCode) !== -1 ?
+            prevLanguageCode : DEFAULT_OPPORTUNITY_LANGUAGE_CODE);
+
+          TranslationLanguageService.setActiveLanguageCode(ctrl.languageCode);
+
+          ctrl.onChangeLanguage = function() {
+            TranslationLanguageService.setActiveLanguageCode(ctrl.languageCode);
+            $window.localStorage.setItem(
+              LAST_SELECTED_TRANSLATION_LANGUAGE, ctrl.languageCode);
+          };
+
+          ctrl.showLanguageSelector = function() {
+            if (!ctrl.tabsDetails[ctrl.activeTabName].hasOwnProperty(
+              'customizationOptions')) {
```

Comment:
This is confusing, I think -- at first glance there doesn't seem to be a good rationale for why  languageSelector relates to the existence of customizationOptions.

I suggest you make customizationOptions mandatory for all tabs and add a karma test to iterate over the list of tab details and ensure each tab detail fully follows the correct schema. If there are no customization options, use an empty list instead of omitting the key altogether. (It's generally easier to maintain things when the list of keys is always fixed.)

Reply:
I think you missed the part of this comment about the karma test, which is also important.